src/lib.rs

```diff
@@ -56,3 +56,14 @@ pub mod scripts;
 pub mod address;
 
 pub mod generated;
+
+/// Static variable defining the current state of execution. The cli binary should
+/// set it to true by calling set_cli_mode.
+#[doc(hidden)]
+pub static IS_CLI_MODE: once_cell::sync::OnceCell<bool> = once_cell::sync::OnceCell::new();
```

Comment:
I know we have `once_cell` as a dependency but is this vastly different to the one in [`std`](https://doc.rust-lang.org/std/cell/struct.OnceCell.html) now?

If that's usable, is it worth using the `std` version instead? In the spirit of eventually removing that dependency.

Reply:
I wanted to minimize change in the PR so I used the tools from the project. I can remove once_cell in this PR too

Reply:
A fair comment: ignore me for now and leave it as-is. We can remove `once_cell` separately (I think it's only used in one other place).

Reply:
I just pushed change removing it :sweat_smile: